From 8d070b3ea765d3e33b2b02dd52c49ee4b70c3252 Mon Sep 17 00:00:00 2001
From: Andre Heinecke <aheinecke@intevation.de>
Date: Wed, 2 Mar 2011 21:27:56 +0100
Subject: [PATCH] Do not use ELEMENT_TYPE which is reserved.

    On Windows Systems ELEMENT_TYPE is defined
    in Winioctl.h so avoid using that name and use
    ELEMENT_CONFIGTYPE instead.
---
 bus/config-parser-common.c |    4 ++--
 bus/config-parser-common.h |    2 +-
 bus/config-parser.c        |    8 ++++----
 3 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/bus/config-parser-common.c b/bus/config-parser-common.c
index f8d75be..c522ff4 100644
--- a/bus/config-parser-common.c
+++ b/bus/config-parser-common.c
@@ -49,7 +49,7 @@ bus_config_parser_element_name_to_type (const char *name)
     }
   else if (strcmp (name, "type") == 0)
     {
-      return ELEMENT_TYPE;
+      return ELEMENT_CONFIGTYPE;
     }
   else if (strcmp (name, "fork") == 0)
     {
@@ -169,7 +169,7 @@ bus_config_parser_element_type_to_name (ElementType type)
       return "servicehelper";
     case ELEMENT_INCLUDEDIR:
       return "includedir";
-    case ELEMENT_TYPE:
+    case ELEMENT_CONFIGTYPE:
       return "type";
     case ELEMENT_SELINUX:
       return "selinux";
diff --git a/bus/config-parser-common.h b/bus/config-parser-common.h
index da74560..b628ea2 100644
--- a/bus/config-parser-common.h
+++ b/bus/config-parser-common.h
@@ -41,7 +41,7 @@ typedef enum
   ELEMENT_SERVICEDIR,
   ELEMENT_SERVICEHELPER,
   ELEMENT_INCLUDEDIR,
-  ELEMENT_TYPE,
+  ELEMENT_CONFIGTYPE,
   ELEMENT_SELINUX,
   ELEMENT_ASSOCIATE,
   ELEMENT_STANDARD_SESSION_SERVICEDIRS,
diff --git a/bus/config-parser.c b/bus/config-parser.c
index 9808d73..2543162 100644
--- a/bus/config-parser.c
+++ b/bus/config-parser.c
@@ -690,12 +690,12 @@ start_busconfig_child (BusConfigParser   *parser,
 
       return TRUE;
     }
-  else if (element_type == ELEMENT_TYPE)
+  else if (element_type == ELEMENT_CONFIGTYPE)
     {
       if (!check_no_attributes (parser, "type", attribute_names, attribute_values, error))
         return FALSE;
 
-      if (push_element (parser, ELEMENT_TYPE) == NULL)
+      if (push_element (parser, ELEMENT_CONFIGTYPE) == NULL)
         {
           BUS_SET_OOM (error);
           return FALSE;
@@ -2002,7 +2002,7 @@ bus_config_parser_end_element (BusConfigParser   *parser,
 
     case ELEMENT_INCLUDE:
     case ELEMENT_USER:
-    case ELEMENT_TYPE:
+    case ELEMENT_CONFIGTYPE:
     case ELEMENT_LISTEN:
     case ELEMENT_PIDFILE:
     case ELEMENT_AUTH:
@@ -2472,7 +2472,7 @@ bus_config_parser_content (BusConfigParser   *parser,
       }
       break;
 
-    case ELEMENT_TYPE:
+    case ELEMENT_CONFIGTYPE:
       {
         char *s;
 
-- 
1.7.0.2.msysgit.0

